Sometimes NoClassDefFoundError happen

When you start MainActivity after logged in to Tumblr, sometimes NoClassDefFoundError happen.

```
09-16 08:45:55.938 17001 17001 I art     : Rejecting re-init on previously-failed class java.lang.Class<com.nomi.artwatch.ui.activity.MainActivity>
09-16 08:45:55.962 17001 17001 E UncaughtException:
09-16 08:45:55.962 17001 17001 E UncaughtException: java.lang.IllegalStateException: Fatal Exception thrown on Scheduler.Worker thread.
09-16 08:45:55.962 17001 17001 E UncaughtException:     at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:62)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at android.os.Handler.handleCallback(Handler.java:739)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at android.os.Handler.dispatchMessage(Handler.java:95)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at android.os.Looper.loop(Looper.java:148)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at android.app.ActivityThread.main(ActivityThread.java:5417)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at java.lang.reflect.Method.invoke(Native Method)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
09-16 08:45:55.962 17001 17001 E UncaughtException: Caused by: java.lang.NoClassDefFoundError: com.nomi.artwatch.ui.activity.MainActivity
09-16 08:45:55.962 17001 17001 E UncaughtException:     at com.nomi.artwatch.ui.util.DeepLinkRouter$Companion.createIntent(DeepLinkRouter.kt:29)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at com.nomi.artwatch.ui.activity.SplashActivity.startDeepLink(SplashActivity.kt:89)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at com.nomi.artwatch.ui.activity.SplashActivity.access$startDeepLink(SplashActivity.kt:21)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at com.nomi.artwatch.ui.activity.SplashActivity$onCreate$subscription$1.call(SplashActivity.kt:47)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at com.nomi.artwatch.ui.activity.SplashActivity$onCreate$subscription$1.call(SplashActivity.kt:21)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at rx.Observable$28.onNext(Observable.java:7976)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at rx.observers.SafeSubscriber.onNext(SafeSubscriber.java:139)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.pollQueue(OperatorObserveOn.java:202)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber$2.call(OperatorObserveOn.java:162)
09-16 08:45:55.962 17001 17001 E UncaughtException:     at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
```

--- a/mobile/src/main/kotlin/com/nomi/artwatch/ui/activity/SplashActivity.kt
+++ b/mobile/src/main/kotlin/com/nomi/artwatch/ui/activity/SplashActivity.kt
@@ -23,6 +23,8 @@ class SplashActivity : InjectActivity() {
     @Inject
     lateinit var mLoginModel: LoginModel
 
+    var mMainIntent: Intent? = null
+
     override fun injectDependency(component: ActivityComponent) {
         component.inject(this)
     }
@@ -30,6 +32,10 @@ class SplashActivity : InjectActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_splash)
+
+        // FIXME：DeepLinkRouter内でIntentを作成すると、たまにNoClassDefFoundErrorが発生するので
+        // MainActivityの起動Intentはここで作成する。
+        mMainIntent = MainActivity.createIntent(this)
 
         if (intent.hasExtra(WatchFaceCompanion.EXTRA_PEER_ID)) {
             Application.setPeerId(intent.getStringExtra(WatchFaceCompanion.EXTRA_PEER_ID))
@@ -86,7 +92,7 @@ class SplashActivity : InjectActivity() {
 
     private fun startDeepLink(uri: Uri) {
         val subscription = DeepLinkRouter
-                .createIntent(this, uri)
+                .createIntent(mMainIntent!!, uri)
                 .subscribe({
                     intent -> startActivity(intent)
                 }, {

--- a/mobile/src/main/kotlin/com/nomi/artwatch/ui/util/DeepLinkRouter.kt
+++ b/mobile/src/main/kotlin/com/nomi/artwatch/ui/util/DeepLinkRouter.kt
@@ -1,10 +1,8 @@
 package com.nomi.artwatch.ui.util
 
-import android.content.Context
 import android.content.Intent
 import android.net.Uri
 import com.nomi.artwatch.Application
-import com.nomi.artwatch.ui.activity.MainActivity
 import com.nomi.artwatch.util.StringUtil
 import rx.Observable
 
@@ -16,7 +14,7 @@ class DeepLinkRouter {
     companion object {
         val LOGIN = "login"
 
-        fun createIntent(context: Context, uri: Uri): Observable<Intent> {
+        fun createIntent(intent: Intent, uri: Uri): Observable<Intent> {
             val host = uri.getHost()
             var path = uri.getPath()
             if (path.startsWith("/")) {
@@ -26,7 +24,10 @@ class DeepLinkRouter {
 
             if (LOGIN.equals(host) && pathParams.size == 1) {
                 // ex. artwatch://login/xxxx
-                val intent = MainActivity.createIntent(context)
+                // FIXME：ここでIntentをつくるとたまにjava.lang.NoClassDefFoundError: com.nomi.artwatch.ui.activity.MainActivity
+                // が発生することがある。
+                // これを避けるために呼び出し側で作成したIntentを渡すようにした
+//                val intent = MainActivity.createIntent(context)
                 Application.setPeerId(pathParams[0])
                 return Observable.just(intent)
 

